[Fuzzing] Add PowerRename Fuzzing and C++ Project Setup Guidance

lei9444 opened the pull request:
<!-- Enter a brief description/summary of your PR here. What does it fix/what does it change/how was it tested (even manually, if necessary)? -->
## Summary of the Pull Request
This PR introduces fuzzing for the PowerRename module to improve reliability by automatically testing its core logic. It also includes setup guidance for other C++ modules that want to adopt a similar fuzzing approach. 

<!-- Please review the items on the PR checklist before submitting-->
## PR Checklist

- [ ] **Closes:** #xxx
- [ ] **Communication:** I've discussed this with core contributors already. If work hasn't been agreed, this work might be rejected
- [ ] **Tests:** Added/updated and all pass
- [ ] **Localization:** All end user facing strings can be localized
- [x] **Dev docs:** Added
- [ ] **New binaries:** Added on the required places
   - [ ] [JSON for signing](https://github.com/microsoft/PowerToys/blob/main/.pipelines/ESRPSigning_core.json) for new binaries
   - [ ] [WXS for installer](https://github.com/microsoft/PowerToys/blob/main/installer/PowerToysSetup/Product.wxs) for new binaries and localization folder
   - [ ] [YML for CI pipeline](https://github.com/microsoft/PowerToys/blob/main/.pipelines/ci/templates/build-powertoys-steps.yml) for new test projects
   - [ ] [YML for signed pipeline](https://github.com/microsoft/PowerToys/blob/main/.pipelines/release.yml)
- [ ] **Documentation updated:** If checked, please file a pull request on [our docs repo](https://github.com/MicrosoftDocs/windows-uwp/tree/docs/hub/powertoys) and link it here: #xxx

<!-- Provide a more detailed description of the PR, other things fixed or any additional comments/features here -->
## Detailed Description of the Pull Request / Additional comments
1. A fuzzing test was created to target CPowerRenameRegEx, aiming to validate its regex parsing and renaming logic under various edge-case inputs. And how to create this project is added here doc/Fuzzing/CppFuzzingGuide.md
2. Add a doc about how to add a fuzzing test for any C++ module
3. Update OneFuzz pipeline to submit the fuzzer to run cloud job
<!-- Describe how you validated the behavior. Add automated tests wherever possible, but list manual validation steps taken as well -->
## Validation Steps Performed
fuzzing pipeline: https://microsoft.visualstudio.com/Dart/_build/results?buildId=120692331&view=results
completed job: https://onefuzz-ui.microsoft.com/jobs/1d628324-0b66-407e-a64d-50e66874fabe
![image](https://github.com/user-attachments/assets/efce7f58-a4cf-4e7d-9302-8d10d95a7bec)

lei9444 committed:
add fuzz

lei9444 committed:
update solution

lei9444 committed:
update pipeline

lei9444 committed:
update solution

lei9444 committed:
remove arm64

lei9444 committed:
use reference

lei9444 committed:
revert the code

lei9444 committed:
add fuzzing readme

lei9444 committed:
Merge branch 'main' of https://github.com/microsoft/PowerToys into leilzh/fuzzcpp

lei9444 committed:
update solution

lei9444 committed:
fix spell-check

lei9444 committed:
Parent reference don't need update

lei9444 committed:
remove fuzzing config

lei9444 commented:
/azp run

yeelam-gordon commented on doc/Fuzzing/CppFuzzingGuide.md:
````diff
@@ -0,0 +1,117 @@
+
+```text
+_DISABLE_VECTOR_ANNOTATION;_DISABLE_STRING_ANNOTATION
+```
````
Should we make these as common fuzzing props and targets for vcxproj?

lei9444 replied:
I will do it in next PR

yeelam-gordon reviewed (approved):
I remember in demo meeting, you talk about:
a. There is a new Configuration in Solution
b. All the fuzz's need dll (i.e. powerRename itself) need also compile with the special option?

lei9444 committed:
add debug config

lei9444 committed:
update the config

lei9444 commented on src/modules/powerrename/PowerRename.FuzzingTest/OneFuzzConfig.json:
```diff
@@ -0,0 +1,40 @@
+        "org": "microsoft",
+        "project": "OS",
+        "AssignedTo": "leilzh@microsoft.com",
+        "AreaPath": "OS\\Windows Client and Services\\WinPD\\DEEP-Developer Experience, Ecosystem and Partnerships\\SHINE\\PowerToys",
```
updated

lei9444 commented on src/modules/powerrename/PowerRename.FuzzingTest/PowerRename.FuzzingTest.cpp:
```diff
@@ -0,0 +1,58 @@
+        return wide;
+    };
+
+    std::wstring input = read_wstring(100);
```
I changed the code to use all, but the string splitting is now more random.

lei9444 commented on src/modules/powerrename/PowerRename.FuzzingTest/OneFuzzConfig.json:
```diff
@@ -0,0 +1,40 @@
+        "AreaPath": "OS\\Windows Client and Services\\WinPD\\DEEP-Developer Experience, Ecosystem and Partnerships\\SHINE\\PowerToys",
+        "IterationPath": "OS\\Future"
+      },
+      "jobNotificationEmail": "leilzh@microsoft.com",
```
changed

lei9444 commented:
> I remember in demo meeting, you talk about: a. There is a new Configuration in Solution b. All the fuzz's need dll (i.e. powerRename itself) need also compile with the special option?

Yes, but after testing, I found that we don’t need to update the reference project. So I removed the fuzzing config and used the release build only. This makes it safer to merge the PR and ensures it won’t affect other projects. I also added a debug config without libFuzzer to make it easier to test the code.

moooyo commented:
LGTM

chenmy77 reviewed (approved):
Everything looks solid.
